### PR TITLE
Update c and cm relaxation tests

### LINKED ISSUE #8: GNU Assembler does NOT support relaxation
If the immediate offset is **less than 4**, the following instructions relaxes to incorrect encodings: 
- c.lbu relaxes to cm.lbu
- c.lhu relaxes to cm.lhu
- c.lh relaxes to cm.lh
- c.sb relaxes to cm.sb
- c.sh relaxes to cm.sh

"If uimm < 4 the encoding is designated for custom use, as the functionality overlaps with
c.lbu: Load unsigned byte, 16-bit encoding."

It is illegal to have the cm.* instructions to execute if the immediate offset is less than 4 as this is designed for custom use.

### PR DESCRIPTION
**Pull request by NandniJamnadas**
Signed-off-by: NandniJamnadas <nandni.jamnadas@embecosm.com>

### CODE REVIEW

File: gas/testsuite/gas/riscv/zc-halfword-test.d [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+#source: zc-halfword-test.s
+#objdump: -d -Mno-aliases
+
+.*:[ 	]+file format .*
+
+
+Disassembly of section .text:
+
+0+000 <target>:
+[  ]+0:[	]+0007c403[ 	]+lbu[	   ]+s0,0\(a5\)
```
Comment (pietraferreira) [inline gas/testsuite/gas/riscv/zc-halfword-test.d]: It seems like the first 4 lines here have an extra space.

File: gas/testsuite/gas/riscv/zc-norm-cm-relax-immu.d [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+#source: zc-norm-cm-relax-immu.s
+#objdump: -d -Mno-aliases
+
+.*:[ 	]+file format .*
+
+
+Disassembly of section .text:
+
+0+000 <target>:
+[  ]+0:[	   ]+8380[ 	      ]+c.lbu[	   ]+s0,0\(a5\)
```
Comment (pietraferreira) [inline gas/testsuite/gas/riscv/zc-norm-cm-relax-immu.d]: Lines 11 to 17, has extra space

File: gas/testsuite/gas/riscv/zc-norm-cm-relax-pass.d [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+.*:[ 	]+file format .*
+
+
+Disassembly of section .text:
+
+0+000 <target>:
+[   ]+0:[	]+23c2[ 	]+cm.lbu[	]+s0,4\(a5\)
+[   ]+2:[	]+2782[ 	]+cm.lbu[	]+s0,8\(a5\)
+[   ]+4:[	]+2fe2[ 	]+cm.lbu[	]+s0,15\(a5\)
+[   ]+6:[	]+33c0[ 	]+cm.lh[	]+s0,4\(a5\)
```
Comment (pietraferreira) [inline gas/testsuite/gas/riscv/zc-norm-cm-relax-pass.d]: Spaces are inconsistent here too, might be worth running `ls -A` on the file to check all the tabs and spaces.

**Comment by pietraferreira** [comment]
I added some comments regarding spaces/tabs as they are inconsistent in some tests. Also, it is probably a good idea to squash the ChangeLog commit, otherwise all good!

File: gas/testsuite/ChangeLog.COREV [hunk 4]
```diff
@@ -1,3 +1,31 @@
...
+	* gas/testsuite/gas/riscv/zc-norm-cm-relax-immu.d: Added 
+	tests to check zcmb do NOT relax riscv instructions to cm 
+	equivalent.
+	* gas/testsuite/gas/riscv/zc-norm-cm-relax-immu.s: Likewise.
+	* gas/testsuite/gas/riscv/zc-norm-cm-relax-pass.d: Added 
+	tests to check zcmb do relax riscv instructions to cm 
+	equivalent within their boundaries.
+	* gas/testsuite/gas/riscv/zc-norm-cm-relax-pass.s: Likewise.
+	
+
```
Comment (CharKeaney) [inline gas/testsuite/ChangeLog.COREV]: Extra line

File: gas/testsuite/ChangeLog.COREV [hunk 5]
```diff
@@ -1,3 +1,46 @@
...
+	* gas/testsuite/gas/riscv/csr-version-1p11.l: Likewise.
+	* gas/testsuite/gas/riscv/csr-version-1p12.d: Likewise.
+	* gas/testsuite/gas/riscv/csr-version-1p12.l: Likewise.
+	* gas/testsuite/gas/riscv/csr-version-1p9p1.d: Likewise.
+	* gas/testsuite/gas/riscv/csr-version-1p9p1.l: Likewise.
+	* gas/testsuite/gas/riscv/csr-version-zcmt.d: Likewise.
+	* gas/testsuite/gas/riscv/csr-version-zcmt.l: Likewise.
+	* gas/testsuite/gas/riscv/csr.s: Likewise.
+
+
```
Comment (CharKeaney) [inline gas/testsuite/ChangeLog.COREV]: Extra line

**Pull request merged**